packages/block-library/src/image/edit.js

```diff
@@ -246,6 +247,11 @@ class ImageEdit extends Component {
 		};
 	}
 
+	getFilename( url ) {
+		const fileName = url.match( /.*\/(.+?)(?:\?.*)?$/ );
```

Comment:
The main reason I'd see to author this way vs. directly return the value (`return url.match( /* ... */ )[ 1 ]`) would be to have the variable serve as an [explaining variable](https://refactoring.com/catalog/extractVariable.html), but in this case its not well-explained in that the return of `String#match` is not the file name, it's a [match `Array`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match#Return_value).

Reply:
This could be made more durable (especially the query parameter) bit if we were to rely on the new `getPath` from `@wordpress/url`.

```js
const path = getPath( url );
if ( path ) {
	return last( path.split( '/' ) );
}
```

Reply:
To our previous conversation about the certainty of the file name being matched; is there any guarantee that the input here is _in-fact_ a `url`? It's making me a bit nervous.